### PR TITLE
[CI] Speed up CI test and fix flaky test

### PR DESCRIPTION
**Pull request by yzang2019**
### Motivation

KoP test is too slow, and there are couple of test very flaky and always timeout.

### Modifications

Parallelize KoP uni tests with multiple jobs after regroup the tests into corresponding packages
Add a testing matrix for github workflow to spin up one job for each group of tests
Increase timeout to fix some flaky tests

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: tests/src/test/java/io/streamnative/pulsar/handlers/kop/TransactionTest.java [hunk 1]
```diff
@@ -93,25 +93,25 @@ protected static Object[][] produceConfigProvider() {
         };
     }
 
-    @Test(timeOut = 1000 * 10, dataProvider = "produceConfigProvider")
+    @Test(timeOut = 1000 * 30, dataProvider = "produceConfigProvider")
```
Comment (yzang2019) [inline tests/src/test/java/io/streamnative/pulsar/handlers/kop/TransactionTest.java:96]: I've run these locally and it takes 22 seconds

File: tests/src/test/java/io/streamnative/pulsar/handlers/kop/MetricsProviderTest.java [hunk 2]
```diff
@@ -314,7 +314,7 @@ public void testUpdateGroupId() {
         });
     }
 
-    @Test(timeOut = 20000, expectedExceptions = KeeperException.NoNodeException.class)
+    @Test(timeOut = 60000, expectedExceptions = KeeperException.NoNodeException.class)
```
Comment (yzang2019) [inline tests/src/test/java/io/streamnative/pulsar/handlers/kop/MetricsProviderTest.java:317]: Tested this locally which takes usually 20-30 seconds to finish

**Review by Demogorgon314 — DISMISSED** [review]

**Comment by BewareMyPower** [comment]
I don't think it speeds up the CI test actually.

image

You can see the `module-test`, i.e. the tests under the `tests/` directory, took the most of the time of CI (44 minutes), while the other tests only took 14 minutes in summary.

See another CI example in https://github.com/streamnative/kop/pull/1914:

image

How could you say this PR speeds up the CI test?

**Comment by gaoran10** [comment]
> I don't think it speeds up the CI test actually.

Maybe we need to add some test groups for tests and use different ci jobs to test various test groups.

**Comment by BewareMyPower** [comment]
> Maybe we need to add some test groups for tests 

+1

**Comment by yzang2019** [comment]
> I don't think it speeds up the CI test actually.
> 
> image
> You can see the `module-test`, i.e. the tests under the `tests/` directory, took the most of the time of CI (44 minutes), while the other tests only took 14 minutes in summary.
> 
> See another CI example in #1914:
> 
> image
> How could you say this PR speeds up the CI test?

Yeah, you are right, I only realized this after I split it into multiple jobs, I think next step is to split them into groups

**Comment by yzang2019** [comment]
@BewareMyPower @Demogorgon314 I've addressed comments, total test time dropped from 50 minutes to around 10 minutes now, could you take another look?

File: tests/src/test/java/io/streamnative/pulsar/handlers/kop/e2e/BasicEndToEndKafkaTest.java [hunk 3]
```diff
@@ -60,27 +62,25 @@ public BasicEndToEndKafkaTest() {
...
+        return new Object[][]{{true}, {false}};
     }
 
     @Test(timeOut = 20000)
     public void testNullValueMessages() throws Exception {
         final String topic = "test-produce-null-value";
 
-        @Cleanup
-        final KafkaProducer<String, String> kafkaProducer = newKafkaProducer();
+        @Cleanup final KafkaProducer<String, String> kafkaProducer = newKafkaProducer();
```
Comment (gaoran10) [inline tests/src/test/java/io/streamnative/pulsar/handlers/kop/e2e/BasicEndToEndKafkaTest.java:72]: Could we retain the code format?

**Review by gaoran10 — COMMENTED** [review]
Great work! LGTM, left a trivial comment.

**Review by BewareMyPower — APPROVED** [review]
Great work!

**Review by Demogorgon314 — APPROVED** [review]

**Review by gaoran10 — APPROVED** [review]

**Pull request merged**